fabric-common-tags-api-v1/src/main/java/net/fabricmc/fabric/api/tags/v1/CommonBlockTags.java

```diff
@@ -0,0 +1,25 @@
+package net.fabricmc.fabric.api.tags.v1;
+
+import net.minecraft.block.Block;
+import net.minecraft.tag.TagKey;
+
+import net.fabricmc.fabric.impl.v1.TagRegistration;
+
+public class CommonBlockTags {
+	public static final TagKey<Block> IRON_ORES = register("iron_ores");
```

Comment:
How does this tag differ from `minecraft:iron_ores`?

Reply:
Shouldn't mods be using `c:iron_ores`? The MC classification tags shouldn't be used by mods imo.

Reply:
I think vanilla tags should be used where they exist, that's what they're for.

Reply:
That leads to inconsistencies which ultimately annoy the modder. E.g. I can't just format `c:%s_ores` if I need to consider that some materials such as tin are in the `c` namespace but others are in the `minecraft` one. Or consider the copper nuggets tag. Minecraft doesn't have them so they should be under `c:`, yet by this logic copper ores should be under the MC namespace? It would be easier for all things to be under the same namespace.

Reply:
Why use namespaces at all if they shouldn't be used? Vanilla Minecraft introduces the concept of copper ores, so its copper ore tags should be used.

Reply:
I agree with Tech, can't you mirror a vanilla tag into the `c:` namespace anyways? 
Only problem is that it might break datapack compat for those that look in `minecraft:` stuff in tags but it doesn't get a mod's. 
Responding to haykam, if some random tech mod added uranium ore, they should add it to c:uranium as that would be the best convention. FAPI should make Minecraft do the same, not special case.

Reply:
As I see it, using `minecraft:` is less desirable than `c:` in that some vanilla tags, like the fluid tag `minecraft:water` also imply functionality that a fluid might not have, such as waterlogging. This leads to awkward places where some vanilla tags are wrapped in the `c:` namespace, but others are not. 

The goal of the tag module (to me) is to aid in mod compact - consistency and discoverability of tags should be higher priority than mimicking vanilla.

Now, including say a `VANILLA_TAGS = net.mc.tag.BlockTags` in `CommonBlockTags` may help alleviate the discoverability issue, but isn't a very good solution.

Yes, we can mirror `minecraft:` into `c:` or visa versa, to avoid accidentally implying that something has functionality that it doesn't, the former is preferable.

Reply:
Mirroring tags to match each other is a mess. Always prefer the standard `minecraft`-namespaced tag if possible, since Minecraft is the true standard and not Fabric API. Mods should be following Minecraft's standards and not Fabric API's standards.

Reply:
If I remember correctly, `c` namespace is a community standard from the datapack community, not FAPI's.

Reply:
It doesn't matter which part of the modding community these tags come from. All that matters is that it deliberately ignores vanilla Minecraft's tags.

Reply:
Vanilla is a standard, but not the standard in modding. I'd rather go with what everyone is using as opposed to what vanilla is alone in doing.

Reply:
Then we should strip the entirety of Minecraft's registries and only have AIR being there, and if you want Vanilla stuff you would need a mod to introduce back the features.

This is a game with a standard, not a game engine like Minetest.

Reply:
Established community standards supercede vanilla's

Reply:
> Established community standards supercede vanilla's

Ok, time to re-introduce a shit ton of stuff that was established community standards but got superseded by Vanilla.

Let's see... time to re-introduce the `biomes` directory for biome tags, or
oh... wait, #1371 was one of those and got widely rejected.

Reply:
If this is what you mean for biomes, datagen does that:
![image](https://user-images.githubusercontent.com/4873274/158075419-13f6ac5a-2e10-4d03-ad2c-e6d6ff4bf629.png)

Reply:
> 

Those will **not** load in 1.18.2 as the sub directory is now `worldgen/biome` and not `biomes`.

That's what I meant, "oh no Mojang introduced biome tags, let's change back the subdirectory to `biomes` because it was community standard then!".

Reply:
The path issue was already fixed in 4695bed91b22f171d3755cd08140fdcddbd7ef94.

Reply:
> The path issue was already fixed in [4695bed](https://github.com/FabricMC/fabric/commit/4695bed91b22f171d3755cd08140fdcddbd7ef94).

That's not my point damn it.

Reply:
Let's not get side-tracked in philosophical concerns here.

Using vanilla tags for existing vanilla items, and c tags for modded items, is more work for modders than going for consistency. So either `minecraft:` only or `c:` only for things such as ingots, ores, nuggets, etc... Imo it is better to use `c:` for everything, than the `minecraft` namespace for everything. This also happens to be the existing status quo, which this module is meant to reflect.

Reply:
Honestly, I am not aware of any mods that use `c` substitutes for vanilla Minecraft's tags, since doing so would be detrimental to mods that properly use the vanilla Minecraft tag.

Namespaces exist for differentiation; as such, it is perfectly fine to mix namespaces. Mods already use their own namespaces, and vanilla Minecraft could be considered the ultimate mod to provide compatibility for.

Reply:
MI uses c tags for everything related to materials, even vanilla tags, and I would like this PR to reduce the amount of special casing needed. Adding materials to vanilla tags is quite bad imo, because some may assume that they only contain vanilla blocks/items. Astromine used to add a few iron ore variants for example, and adding them to the minecraft tag might cause issues.

Reply:
> Let's not get side-tracked in philosophical concerns here.

I fail to see what the "philosophical concerns" are out of this

The c:*_ores standard is from before Minecraft added its own tags for this, making this useless

Reply:
> Adding materials to vanilla tags is quite bad imo, because some may assume that they only contain vanilla blocks/items.

This assumption is completely wrong and should not be considered. Just like tags added by mods, vanilla Minecraft's tags can support members from any namespace, not just their own. Otherwise, they would be completely useless for compatibility.

Reply:
> Adding materials to vanilla tags is quite bad imo, because some may assume that they only contain vanilla blocks/items.

This would be an incredibly dumb assumption given tags like minecraft:mineable/axe imply no such guarantees

Reply:
> Let's not get side-tracked in philosophical concerns here.
> 
> Using vanilla tags for existing vanilla items, and c tags for modded items, is more work for modders than going for consistency. So either `minecraft:` only or `c:` only for things such as ingots, ores, nuggets, etc... Imo it is better to use `c:` for everything, than the `minecraft` namespace for everything. This also happens to be the existing status quo, which this module is meant to reflect.

Hello? Are you stuck in 1.16?

This PR is for 1.18, the status quo is the minecraft tags.

Reply:
Why not have a static method like `CommonBlockTags.ore("iron")` that uses the correct namespace?

Reply:
Hi, just wanted to provide my thoughts here as a relatively new Fabric modder. I recently ported Scannable [to Fabric](https://github.com/embeddedt/Scannable-Fabric) and it made use of the `forge:ores/iron`, etc. tags to find ore blocks. When I had to port this functionality to Fabric, I found the current tag situation to be very confusing. Some mods add the `minecraft:iron_ore` block to `#c:iron_ores`, some mods add the vanilla tag to `#c:iron_ores`, and some do nothing. That means that the behavior would be different depending on whether those mods are installed or not.

My solution at the moment was to have Scannable always add the vanilla tag to the `c` tag in datagen and then use the `c` tag everywhere, as I figured that would work with existing mods... but I agree with @Technici4n that having two different namespaces makes no sense and creates more confusion. I think it should be decided whether the standard is to add to the vanilla tag or add to the `c` tag and include the vanilla tag so vanilla ores are also part of the `c` tag.

Reply:
Why is copying into the `c` tag the only option? We could just as easily copy `c` into `minecraft`, which is arguably more correct as vanilla is providing an existing standard

Reply:
Or... just use the `minecraft` namespace from the start?

Reply:
So `minecraft:iron_ores` but `c:iron_ingots`, `c:iron_dusts` and `c:tin_ores`? That seems very inconsistent to me for these classification tags, and I would prefer everything under the same namespace.

Reply:
> So `minecraft:iron_ores` but `c:iron_ingots`, `c:iron_dusts` and `c:tin_ores`? That seems very inconsistent to me for these classification tags, and I would prefer everything under the same namespace.

Yeah, then have 2 redundant tags that definitely won't cause issues down the line.
Who cares one is in the minecraft one and one isn't?
HEY, it's even clearer as it makes it clear from the start who it belongs to!

*looks*
oh! this is a tag provided by Minecraft, it will always be here!
oh! this is a tag provided by the community, not always be here!

Reply:
It is a matter of consistency for the user, I prefer that even if it entails some redundancy. However, it is true that it may cause issues with datapacks or mods that don't rely on fabric API being present. I don't know if it is possible to mirror two tags by including each one in the other, but that sounds like the best option imo. Otherwise, having the common one include the MC one would work too.

Not sure I follow your other point... It seems to me that the whole point of this module is to provide tags that will "always" be available.

Reply:
You cannot have cyclic tags.

And no, tags are not always available if this module is not present.
And in contrary, it's less consistent as now for a datapacker there's 2 equivalent tags, time to figure out which one to edit without breaking everything.

If you want in-code consistency, it's called constants variables.

Reply:
> (but never add any block directly to the common tag)

You cannot stop mods from doing that.

> I am formatting the tag based on the material and the part type

But you should be formating for both minecraft and c namespaces anyways? People use the Minecraft tags because they already exist, you're limiting compatibility by ignoring them for some weird idea of purity you seem to have. MI is not a standard in Minecraft or the wider community by any stretch

Reply:
> I am not using constant variables, I am formatting the tag based on the material and the part type. I will keep using the c tag in any case because it makes my life simpler, but I think having it in the module and defining to which tag blocks should be added to is the easiest: less work for modders in a situation similar to mine.

It stays useless and only has inconveniences:
 - confuses modders and datapackers with 2 tags with the same content
 - clutters the debug screen
 - duplicates information, it's most often considered a very bad thing since it can quickly go wrong
 - Minecraft is still the only source of truth
 - extra maintenance burden, it's even usually your argument

At this point we probably should also register `c:azalea_log` or `c:redwood_log` blocks too since they are widely used and not in the same namespace which oh no complicates stuff!!!

Reply:
I agree that there needs to be one namespace for vanilla and modded ores, otherwise it's not possible to predictably generate the tag name given just the material and metal type, without needing to special-case the vanilla ones. I think the main question is whether creating a `#minecraft:tin_ores` tag when that doesn't exist in vanilla is acceptable.

Reply:
Creating a non vanilla tag in the minecraft namespace is unacceptable.

Reply:
What is it with everyone and string formatting for functionality???

That is not reliable and easily broken by locale, *please* come up with some reasonable system, this is insane

Reply:
```java
String material = "iron"; // or "gold", etc
new ResourceLocation("c", material + "_ores")
```

Maybe I'm missing something, but I don't see why this wouldn't work in any locale.

Reply:
That's not "predictably generating" anything, in fact that'll be constant folded. As soon as you need to do that for arbitrary materials you either run into a massive conditional, or having to work around the unlimited ways things can be referenced

Reply:
As someone who is actively working on a mod (Aurora's Decorations) whose goal is to auto generate as much as possible, I can assure you that namespace is not an issue, and I'm getting really tired of the tantrum from the tech mod dev.

What is suggested here is a dangerous precedent, start doing this with tags and you will not have justification left to not do this with common modded blocks, items, entities too.

Some of the tags here can be useful (food tag but only if auto-generated at runtime, could be cool for some recipes and stuff, or the biome ones as it was justified even though not entirely satisfied with them), but the ore ones along with many others are straight up stupid, dangerous, etc. I am tired of repeating myself. We already surpassed the amount of comments of #1371, this is a sad milestone.

Reply:
> That's not "predictably generating" anything, in fact that'll be constant folded. As soon as you need to do that for arbitrary materials you either run into a massive conditional, or having to work around the unlimited ways things can be referenced

I'm not sure about what you mean here, MI does exactly this for all of its material recipes and it works fine, quite predictably imo.

> I can assure you that namespace is not an issue

It is possible to work around it (for example by introducing the `c:` tag in one's own mod), but it's quite annoying and could be done once by Fabric instead of requiring multiple mods to do it.

> I'm getting really tired of the tantrum from the tech mod dev

I personally don't see the issue with fabric being somewhat tech mod-friendly. You may not see the appeal of this feature, but multiple people have told you that it is useful, and there is no need to insult them.

Reply:
I have nothing against Fabric being tech mod-friendly, my issue is when non-sense is pushed.

As I said, if we start doing this, then why not have a `c:tin_ingot` item at this point since multiple mods add it, voilà no more name/identifier/namespace issue!
My issue is we're literally arguing over tape just because some people want all of their tape red and refuse to mix the black tape in.

Meanwhile I'm here with a taped canvas of every color imaginable.
~~And a screw Fabric continuously refuses to use because it required extra work on a non-Fabric-managed mod.~~

I also think you're terribly misplaced to tell someone to have respect. And if I follow some standards, this was no insult.

Reply:
> I'm not sure about what you mean here, MI does exactly this for all of its material recipes and it works fine, quite predictably imo.

Then none of this is "predicatively generated", its hardcoded.

Reply:
> Then none of this is "predicatively generated", its hardcoded.

Well it is "hardcoded" over the cartesian product of the materials and the parts, but I wouldn't really call this hardcoded:
- https://github.com/AztechMC/Modern-Industrialization/blob/15e36c35c54ec8ddd066c8533a24f8c1d12b93d2/src/main/java/aztech/modern_industrialization/materials/part/RegularPart.java#L202.
- The only exception is raw metals, for which the conventional tags are `c:raw_%s_ores` and `c:raw_%s_blocks`, so they need some special casing: https://github.com/AztechMC/Modern-Industrialization/blob/15e36c35c54ec8ddd066c8533a24f8c1d12b93d2/src/main/java/aztech/modern_industrialization/materials/part/RawMetalPart.java#L71.

> As I said, if we start doing this, then why not have a c:tin_ingot item at this point since multiple mods add it, voilà no more name/identifier/namespace issue!

I don't get your comparison... we are talking about adding tags, not items.

Reply:
Its hardcoded because every material there could be compile time constants, there is nothing dynamic here, you are making an assumption about formatting and forcing it on other mods

Reply:
Also as youve put it

> The point of this PR is to reflect the status quo, and not to try to change it.

Tech mods are not the entire status quo, using them exclusively is not helpful

Reply:
Why was this marked as resolved?
The critics are clearly still here.

Reply:
because it was removed?

Reply:
They're gone.

Reply:
They are not. Might wanna consider buying glasses.

Reply:
Update: ok, I'm the one who needs to buy glasses actually.